Framework/Core/src/DecongestionService.h

```diff
@@ -20,6 +20,8 @@ struct DecongestionService {
   bool isFirstInTopology = true;
   /// Last timeslice we communicated. Notice this should never go backwards.
   int64_t lastTimeslice = 0;
+  /// The next timeslice we should consume, when running in order
+  int64_t nextTimeslice = 0;
```

Comment:
makes sense to me, I would make it a bit more clear here that this is only for the ordered completion policy.
What I like about this in particular is that we can reset them together for START/STOP/START, and we don't need an extra reset for the completion policy.